src/shims/posix/fs.rs

```diff
@@ -924,6 +981,22 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             )
         }
 
+        // Reject if isolation is enabled.
+        if let IsolatedOp::Reject(reject_with) = this.machine.isolated_op {
+            this.reject_in_isolation("`statx`", reject_with)?;
+            let ecode = if path.is_absolute() || dirfd == this.eval_libc_i32("AT_FDCWD")? {
+                // since `path` is provided, either absolute or
+                // relative to CWD, `EACCES` is the most relevant.
+                this.eval_libc("EACCES")?
+            } else {
+                // `dirfd` is set to target file, and `path` is
+                // empty. `EACCES` would violate the spec.
```

Comment:
Why is path empty? It could be relative, no?

Reply:
`if` block handles the case of `path` being relative. In that case, dirfd has to be set to `AT_FDCWD`.

Reply:
```suggestion
                // empty (or we would have hit the `throw_unsup_format` above). `EACCES` would violate the spec.
                assert!(empty_path_flag);
```
Just making sure we notice if the code above ever changes and someone forgets to update this code here.

Reply:
Yeah, I was also not completely happy repeating the valid input checks. One more option is, I can combine two if statements into one, avoiding code duplication.

Reply:
No I think I like the code organization here. Just please add the `assert!`.

Reply:
Alright, done.